test/external/channel-benchmark/CMakeLists.txt

```diff
@@ -0,0 +1,71 @@
+include(NeuronTestHelper)
+set(prefix "benchmark/channels") # everything is under this prefix in the repository
+set(channel_benchmark_sim_time 10)
+set(channel_benchmark_mpi_ranks 1)
+set(channel_benchmark_neuron_prefix
+    OMP_NUM_THREADS=1
+    "HOC_LIBRARY_PATH=${prefix}/lib/hoclib"
+    ${MPIEXEC_NAME}
+    ${MPIEXEC_NUMPROC_FLAG}
+    ${channel_benchmark_mpi_ranks}
+    ${MPIEXEC_OVERSUBSCRIBE}
+    ${MPIEXEC_PREFLAGS}
+    special
+    ${MPIEXEC_POSTFLAGS})
+
+set(channel_benchmark_hippo_num_cells 6) # Should be at least 6 to cover all cell types
+set(channel_benchmark_sscx_num_cells 16) # Similarly, should be at least 16.
+set(channel_benchmark_gpu_args -c arg_coreneuron_gpu=1)
+set(channel_benchmark_coreneuron_gpu_args --gpu --cell-permute=2)
+foreach(model hippo sscx)
+  nrn_add_test_group(
+    NAME channel_benchmark_${model}
+    SUBMODULE tests/channel-benchmark
+    OUTPUT asciispikes::out.dat
+    MODFILE_PATTERNS "${prefix}/lib/modlib/*.mod"
+    SCRIPT_PATTERNS "${prefix}/lib/hoclib/*.hoc" "${prefix}/${model}/*.txt"
+                    "${prefix}/${model}/morphologies/*.asc")
+  set(channel_benchmark_neuron_args
+      -mpi
+      -c
+      "strdef arg_datapath"
+      -c
+      "arg_datapath = \"${prefix}\""
+      -c
+      arg_target_count=${channel_benchmark_${model}_num_cells}
+      -c
+      arg_tstop=${channel_benchmark_sim_time}
+      -c
+      "strdef arg_model"
+      -c
+      "arg_model=\"${model}\""
+      "${prefix}/lib/hoclib/init.hoc")
+  nrn_add_test(
+    GROUP channel_benchmark_${model}
+    NAME neuron
+    REQUIRES mpi
+    PROCESSORS ${channel_benchmark_mpi_ranks}
+    COMMAND ${channel_benchmark_neuron_prefix} ${channel_benchmark_neuron_args})
+  foreach(processor gpu cpu)
+    nrn_add_test(
+      GROUP channel_benchmark_${model}
+      NAME coreneuron_${processor}
+      REQUIRES coreneuron ${processor}
+      PROCESSORS ${channel_benchmark_mpi_ranks}
+      COMMAND ${channel_benchmark_neuron_prefix} -c arg_coreneuron=1
+              ${channel_benchmark_${processor}_args} ${channel_benchmark_neuron_args})
+    nrn_add_test(
+      GROUP channel_benchmark_${model}
+      NAME coreneuron_${processor}_offline
+      REQUIRES coreneuron mpi ${processor}
+      PROCESSORS ${channel_benchmark_mpi_ranks}
+      PRECOMMAND ${channel_benchmark_neuron_prefix} -c arg_dump_coreneuron_model=1
+                 ${channel_benchmark_${processor}_args} ${channel_benchmark_neuron_args}
+      COMMAND
+        OMP_NUM_THREADS=1 ${MPIEXEC_NAME} ${MPIEXEC_NUMPROC_FLAG} ${channel_benchmark_mpi_ranks}
+        ${MPIEXEC_OVERSUBSCRIBE} ${MPIEXEC_PREFLAGS} special-core ${MPIEXEC_POSTFLAGS} --mpi
+        --tstop ${channel_benchmark_sim_time} --voltage 1000. -d coredat/
+        ${channel_benchmark_coreneuron_${processor}_args})
```

Comment:
could you remind me if the offline mode lets is required? i.e. if we do file-transfer mode then it should be sufficient?

Reply:
I don't think the offline mode test is strictly required. This just follows the pattern of e.g. the `testcorenrn` tests.
For debugging future failures, the fully "offline" mode we use at the moment leaves the input data on disk in a well-defined place, which might be convenient? (`test/channel_benchmark_hippo/coreneuron_cpu_offline/coredat`).

Reply:
> the fully "offline" mode we use at the moment leaves the input data on disk in a well-defined place, which might be convenient? 

if we use `-c arg_coreneuron_filemode=1` then it would have exact same effect but just hiding all the CLI parameters  mentioned in the last `COMMAND` clause?

if that doesn't work, we can merge as it is. Just trying to see if all these ugly params could be reduced.